psu/psu.c

```diff
@@ -760,15 +766,48 @@ static const script_funct_t funcs[] = {
 
 int main(int argc, char *argv[])
 {
-	int res = -1;
+	long int tmp;
+	int opt, res = -1;
 	script_t script;
 	hid_device *dev = NULL;
+	char *ptr;
 
-	if (argc < 2)
-		return usage(argv[0]);
+	for (;;) {
+		opt = getopt(argc, argv, "ht:");
+		if (opt == -1) {
+			break;
+		}
 
-	if (script_load(&script, argv[1]) != SCRIPT_OK)
-		return -1;
+		switch (opt) {
+			case 'h':
+				usage(argv[0]);
+				return EXIT_SUCCESS;
+
+			case 't':
+				tmp = strtol(optarg, &ptr, 10);
+				if ((optarg == ptr) || (tmp < 0) || (tmp > INT_MAX)) {
```

Comment:
You have merged this PR, but you only check invalid first character in the `optarg`.
```c
				if ((*ptr != '\0') || (tmp < 0) || (tmp > INT_MAX)) {
```

Reply:
Right. Added quick fix, but checking `optarg == ptr` is also necessary in case the input was empty `-t ""`, the strtol value returned is then `0`, but there should be an error: invalid value. Thanks.